kratos/includes/kratos_application.h

```diff
@@ -303,25 +349,62 @@ class KRATOS_API(KRATOS_CORE) KratosApplication {
+    //Sphere
+    const Sphere3D1<NodeType> mSphere3D1Prototype = Sphere3D1<NodeType>(GeometryType::PointsArrayType(1));
+    //Lines:
+    const Line2D2<NodeType> mLine2D2Prototype = Line2D2<NodeType>(GeometryType::PointsArrayType(2));
+    const Line2D3<NodeType> mLine2D3Prototype = Line2D3<NodeType>(GeometryType::PointsArrayType(3));
+    const Line3D2<NodeType> mLine3D2Prototype = Line3D2<NodeType>(GeometryType::PointsArrayType(2));
+    const Line3D3<NodeType> mLine3D3Prototype = Line3D3<NodeType>(GeometryType::PointsArrayType(3));
+    //Triangles:
+    const Triangle2D3<NodeType> mTriangle2D3Prototype = Triangle2D3<NodeType>(GeometryType::PointsArrayType(3));
+    const Triangle2D6<NodeType> mTriangle2D6Prototype = Triangle2D6<NodeType>(GeometryType::PointsArrayType(6));
+    const Triangle3D3<NodeType> mTriangle3D3Prototype = Triangle3D3<NodeType>(GeometryType::PointsArrayType(3));
+    const Triangle3D6<NodeType> mTriangle3D6Prototype = Triangle3D6<NodeType>( GeometryType::PointsArrayType(6));
+    //Quadrilaterals:
+    const Quadrilateral2D4<NodeType> mQuadrilateral2D4Prototype = Quadrilateral2D4<NodeType>( GeometryType::PointsArrayType(4));
+    const Quadrilateral2D8<NodeType> mQuadrilateral2D8Prototype = Quadrilateral2D8<NodeType>( GeometryType::PointsArrayType(8));
+    const Quadrilateral2D9<NodeType> mQuadrilateral2D9Prototype = Quadrilateral2D9<NodeType>( GeometryType::PointsArrayType(9));
+    const Quadrilateral3D4<NodeType> mQuadrilateral3D4Prototype = Quadrilateral3D4<NodeType>( GeometryType::PointsArrayType(4));
+    const Quadrilateral3D8<NodeType> mQuadrilateral3D8Prototype = Quadrilateral3D8<NodeType>( GeometryType::PointsArrayType(8));
+    const Quadrilateral3D9<NodeType> mQuadrilateral3D9Prototype = Quadrilateral3D9<NodeType>( GeometryType::PointsArrayType(9));
+    //Tetrahedra:
+    const Tetrahedra3D4<NodeType> mTetrahedra3D4Prototype = Tetrahedra3D4<NodeType>( GeometryType::PointsArrayType(4));
+    const Tetrahedra3D10<NodeType> mTetrahedra3D10Prototype = Tetrahedra3D10<NodeType>( GeometryType::PointsArrayType(10));
+    //Prisms:
+    const Prism3D6<NodeType> mPrism3D6Prototype = Prism3D6<NodeType>( GeometryType::PointsArrayType(6));
+    const Prism3D15<NodeType> mPrism3D15Prototype = Prism3D15<NodeType>( GeometryType::PointsArrayType(15));
+    //Hexahedra:
+    const Hexahedra3D8<NodeType> mHexahedra3D8Prototype = Hexahedra3D8<NodeType>( GeometryType::PointsArrayType(8));
+    const Hexahedra3D20<NodeType> mHexahedra3D20Prototype = Hexahedra3D20<NodeType>( GeometryType::PointsArrayType(20));
+    const Hexahedra3D27<NodeType> mHexahedra3D27Prototype = Hexahedra3D27<NodeType>( GeometryType::PointsArrayType(27));
+    //QuadraturePointGeometry
+    const std::size_t size_quadrature = 1;
+    const QuadraturePointGeometry<NodeType, 3>* mpQuadraturePointGeometryVolume3dPrototype = new QuadraturePointGeometry<NodeType, 3>(GeometryType::PointsArrayType(size_quadrature));
```

Comment:
Not sure if this is a good idea
I think it would be better to solve the original problem
Btw as it is now I think its a memory leak, you are not deleting it

Reply:
No, the problem is already solved. The code giving memory leak was not commited. This one works, we can use references as in the other geometries if we change things in the QuadraturePoints geometry

Reply:
I mean that you allocate a raw ptr with new but never delete it

Could you please explain why the way the other geometries are registered doesn’t work?

Reply:
Does not compile, I don't remember the error, but it was something that some member variable was private

Reply:
Can you post the error?

Reply:
Can it be that it is the same error as in #5977.

There might be something missing in those geometries... However, I have no glue what it is.

Reply:
> Can you post the error?

It is really a problem to use a raw pointer directly?, I can, but I think this can be fixed later (otherwise we will need to modify additional files)

Reply:
In principle I am fine with it but I would like to understand why this is happening 
It is a very important part of the code

Reply:
~~~
/home/vicente/src/Kratos/kratos/geometries/quadrature_point_geometry.h:155:5: error: field of type 'Kratos::GeometryData' has private default constructor
    QuadraturePointGeometry( QuadraturePointGeometry const& rOther )
    ^
/home/vicente/src/Kratos/kratos/includes/kratos_application.h:403:94: note: in instantiation of member function 'Kratos::QuadraturePointGeometry<Kratos::Node<3, Kratos::Dof<double> >, 3, 2, 2>::QuadraturePointGeometry' requested here
    const QuadraturePointGeometry<NodeType, 3, 2> mQuadraturePointGeometrySurface3dPrototype = QuadraturePointGeometry<NodeType, 3, 2>(GeometryType::PointsArrayType(size_quadrature));
                                                                                             ^
/home/vicente/src/Kratos/kratos/geometries/geometry_data.h:749:5: note: declared private here
    GeometryData()
~~~

Reply:
thx

Ok this seems to me like an issue with the assignment constructor
I think it works when using the raw-ptr because there you don't call the assignment constructor
can you try directly calling the constructor?
```suggestion
    const QuadraturePointGeometry<NodeType, 3> mQuadraturePointGeometryVolume3dPrototype(GeometryType::PointsArrayType(size_quadrature));
```

@tteschemacher did you use / make tests for the assignment constructors of the new geometries?

Probably using the assignment constructors works for the other geometries because the `GeometryData` is handled differently there

Reply:
no, I didn't.

Reply:
I think you should definitely add test for copy and assignments. We cannot disallow copying geometries until we find a very good reason for that.

Reply:
This is for me or for @tteschemacher?, and how I test copies and assigments?

Reply:
I will take care about that, especially for the nurbs_geometries.
However, the second question is still interesting?